### PR TITLE
Issue/8339 add history list

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Add a list shown when tapping the ***History*** item in the editor overflow menu as described in https://github.com/wordpress-mobile/WordPress-Android/issues/8339.  See the screenshots below for detail.

revisions_history_list

#### Note
There are multiple differences between the ***Design*** and ***Develop*** screenshots.  The avatar default and error background was added with the [gridicons-user](https://github.com/Automattic/gridicons/blob/master/svg-min/gridicons-user.svg) icon.  The header was updated to use the date rather than today, yesterday, last week, etc. to mimic the ***Activity Log***.  The title text was updated to use the same method and consequently produce the same time ago format as the post status on the ***Blog Posts***.  The subtitle text was updated to use the date and time of the version since the status is not included in the page/post history.  The date text was updated to have uppercase AM/PM since that follows the built-in format from Java based on the device locale.  The divider was updated to be full-width to mimic the ***Activity Log***.

The avatar icon will be updated with the avatar image of the version's author in a subsequent pull request.

#### Site Pages
1. Go to ***My Site*** tab.
2. Tap ***Site Pages*** item under ***Publish*** section.
3. Tap overflow menu button in top toolbar.
4. Tap ***History*** item in menu.
5. Notice ***History*** list as shown above.

#### Blog Sites
1. Go to ***My Site*** tab.
2. Tap ***Blog Sites*** item under ***Publish*** section.
3. Tap overflow menu button in top toolbar.
4. Tap ***History*** item in menu.
5. Notice ***History*** list as shown above.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### CODE REVIEW

**Comment by khaykov** [comment]
@iamthomasbishop I realized that we don't show usernames in Revisions list, unlike Calypso. Do you think that it might be hard to identify users just by avatars?

**Comment by theck13** [comment]
Good point, @khaykov.  I discussed this view with @iamthomasbishop last week and we considered using usernames in place of the version status for the subtitle.  I used the date and time for now since that's the data we have and I plan to update it with whatever we choose in a subsequent pull request.  I think using the display name like Calypso or the username would be helpful especially if the user does not have a avatar and the default icon is shown.

**Comment by iamthomasbishop** [comment]
Especially in cases where a site has multiple authors, a username or full name would probably be more useful. The original design had the post state at the specific point in time, but we could also add the author name – something like this:

`Time ago`
`Post State (Author Name)`

**Comment by theck13** [comment]
Unfortunately, I don't think the post state is saved in the history.  Calypso and the API endpoint don't contain that information.  Neither the author display name nor username are included either at the moment, but we may be able to add that to the API endpoint and include it in a subsequent pull request.

File: WordPress/src/main/java/org/wordpress/android/ui/history/HistoryListItem.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        val postDateGmt: String?,
+        val postModifiedGmt: String?,
+        val postAuthorId: String?,
+        val titleDiffs: ArrayList<Diff>,
+        val contentDiffs: ArrayList<Diff>
+    ) : HistoryListItem(REVISION) {
+        // Replace space with T since API returns yyyy-MM-dd hh:mm:ssZ and ISO 8601 format is yyyy-MM-ddThh:mm:ssZ.
+        private val postDate: Date = DateTimeUtils.dateUTCFromIso8601(postDateGmt?.replace(" ", "T"))
+        val timeSpan: String = DateTimeUtils.javaDateToTimeSpan(postDate, WordPress.getContext())
+        val formattedDate: String = DateFormat.getDateInstance(DateFormat.LONG, Locale.getDefault()).format(postDate)
```
[the remark this answers is not in the document]
Comment (theck13) [reply WordPress/src/main/java/org/wordpress/android/ui/history/HistoryListItem.kt]: I updated the formatted date value in 6658b12.

File: WordPress/src/main/java/org/wordpress/android/ui/history/HistoryListItem.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        val formattedTime: String = DateFormat.getTimeInstance(DateFormat.SHORT, Locale.getDefault()).format(postDate)
```
[the remark this answers is not in the document]
Comment (theck13) [reply WordPress/src/main/java/org/wordpress/android/ui/history/HistoryListItem.kt]: I updated the formatted time value in 2248f8c.

**Comment by theck13** [comment]
> When I was testing I noticed that the new page/post has the History menu item enabled. When you open it, there is only a blank screen (not even an empty state).

You're right.  New pages and posts don't have history and shouldn't show that menu item.  I added a condition to hide the _*History*_ menu item for new pages and posts in e755099.

**Comment by 0nko** [comment]
> Can you test with that post on develop? I know there's no way to refresh a view in the editor at the moment, but I'm wondering if going to Preview or Post Settings and rotating the device would do the same thing on develop.

I tested it with Preview and the same thing happened, so it's not related to this PR.

**Pull request merged**